onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/cloudide/factory/filter/AgreementPageFilter.java

```diff
@@ -0,0 +1,106 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.cloudide.factory.filter;
+
+import org.eclipse.che.api.factory.shared.dto.Factory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.net.URLEncoder;
+
+/**
+ * Redirects user to the agreement page when using factory for first time.
+ *
+ * @author Max Shaposhnik
+ *
+ */
+@Singleton
+public class AgreementPageFilter implements Filter {
+
+    @Inject
+    @Named("page.creation.error")
+    protected String CREATION_FAILED_PAGE;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void destroy() {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+//        HttpServletRequest httpReq = (HttpServletRequest)servletRequest;
```

Comment:
Remove class if it is not  needed. If it is needed disable it in path mapping instead of commenting it